alcs-frontend/src/app/services/notice-of-intent/notice-of-intent.service.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { HttpClient } from '@angular/common/http';
+import { TestBed } from '@angular/core/testing';
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { firstValueFrom, of, throwError } from 'rxjs';
+import { ToastService } from '../toast/toast.service';
+
+import { NoticeOfIntentService } from './notice-of-intent.service';
+
+describe('NoticeOfIntentService', () => {
+  let service: NoticeOfIntentService;
+  let httpClient: DeepMocked<HttpClient>;
+  let toastService: DeepMocked<ToastService>;
+
+  beforeEach(() => {
+    httpClient = createMock();
+    toastService = createMock();
+
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: HttpClient,
+          useValue: httpClient,
+        },
+        {
+          provide: ToastService,
+          useValue: toastService,
+        },
+      ],
+    });
+    service = TestBed.inject(NoticeOfIntentService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should create covenants', async () => {
+    httpClient.post.mockReturnValue(
+      of({
+        fileNumber: '1',
+      })
+    );
+
+    const res = await service.create({
+      applicant: '',
+      boardCode: '',
+      fileNumber: '',
+      localGovernmentUuid: '',
+      regionCode: '',
+    });
+
+    expect(httpClient.post).toHaveBeenCalledTimes(1);
+    expect(res.fileNumber).toEqual('1');
+  });
+
+  it('should show an error toast message if create fails', async () => {
+    httpClient.post.mockReturnValue(
+      throwError(() => {
+        new Error('');
+      })
+    );
+
+    try {
+      await service.create({
```

Comment:
is "expect(await service.create()).rejects.toMatchObject..." is not working?

Reply:
Cleaned up